gem/optimise.py

```diff
@@ -187,6 +187,12 @@ def child(expression):
     if types <= {Literal, Zero, Failure}:
         return partial_indexed(ListTensor(expressions), (index,))
 
+    if types <= {ComponentTensor, Zero}:
```

Comment:
So this function is called on `exprs` which have had component tensors removed (L228-230).  So how can a `ComponentTensor` show up here?